eng/resolveContract.targets

```diff
@@ -12,7 +12,7 @@
     <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and
                                      '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                                      $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">$(NetCoreAppCurrentRefPath)$(TargetFileName)</ContractAssemblyPath>
-    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == ''">$([MSBuild]::NormalizePath('$(BaseOutputPath)', 'ref', '$(TargetFramework)-$(Configuration)', '$(TargetFileName)'))</ContractAssemblyPath>
+    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and !$(TargetFramework.Contains('-'))">$([MSBuild]::NormalizePath('$(BaseOutputPath)', 'ref', '$(TargetFramework)-$(Configuration)', '$(TargetFileName)'))</ContractAssemblyPath>
```

Comment:
What are you trying to achieve here?

Reply:
I added this check to avoid setting the wrong value for ContractAssemblyPath in cases like net5.0-windows. (we dont have ant net5.0-platform) currently.

Reply:
This forces references assemblies to be (re-)built if the consuming TargetFramework contains the delimiter. This would be a huge regression in the developer innerloop.

Reply:
> This forces references assemblies to be (re-)built if the consuming TargetFramework contains the delimiter. This would be a huge regression in the developer innerloop.

for net6.0, we will not be using this line as Contract Assembly will already be initialized by line 14, for other targetFrameworks(except net5.0) we are removing the platform from the tfm. 
We dont have any project that has a net5.0 platform specific configuration. if in the future we add such a config we can just parse the tfm here and initialize the contract assembly.

Reply:
> if in the future we add such a config we can just parse the tfm here and initialize the contract assembly.

what do you mean by "just parse the tfm here and initialize the contract assembly"?

Reply:
so if we add a project that target net5.0-Windows, then we will need to add
```
<TargetFrameworkWithoutPlatform>$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', '(-[^;]+)', ''))</TargetFrameworkWithoutPlatform>
<ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and $(TargetFramework.Contains('-'))">$([MSBuild]::NormalizePath('$(BaseOutputPath)', 'ref', '$(TargetFrameworkWithoutPlatform)-$(Configuration)', '$(TargetFileName)'))</ContractAssemblyPath>
```

Reply:
> so if we add a project that target net5.0-Windows, then we will need to add

How would you track that someone adds such a configuration in their PR to react to it with infra support?

Reply:
To close on this, you understand my point? I think we shouldn't layer our infra on what's existent but what we support. Also covers reactive vs proactive work.

Reply:
>To close on this, you understand my point? I think we shouldn't layer our infra on what's existent but what we support. Also covers reactive vs proactive work.

yeah i understood where we would have caused a regression and not caught it.